### DISCUSSION TITLE
How to exclude unrelated commit tags when extracting part of the repository

### DISCUSSION THREAD

**Discussion by Rgzevs** [newren/git-filter-repo#200] [Q&A] [answered]
I've inherited a repository that contains several unrelated projects and am trying to split it using `git-filter-repo`. 
My problem is that while filtering repository by folder the output  contains tags from the commits that didn't affect the selected folder. For example the repository contains folders A and B and the commits look like this:

1. Files in folder A, tag:<1>
2. Files in  folder B, tag:<2>
3. Files in folder B, tag:<3>
4. Files in folder A, tag:<4>

So if I want to filter only commits from folder A, I'm expecting the resulting commits to be like this:

1. Files in folder A, tag:**<1>**
4. Files in folder A, tag:**<4>**

But when I run  `git-filter-repo --subdirectory-filter A --replace-refs delete-no-add ` instead the result looks like this:

1. Files in folder A, tag:**<1>**
4. Files in folder A, tags:**<4>,<2>,<3>**

So it looks like all tags from the commits that have been filtered out are added to the next commit that fits the filter.
Is it possible to drop all the unrelated tags from `git-filter-repo` output?

**Comment by newren** [comment, marked as answer]
When filter-repo removes commits because they have no more logical changes, any tags pointing to those commits need to be rewritten to the nearest ancestor.  So, the case you describe above suggest that tags <2> and <3> pointed to commits that contained the history to which <4> more directly pointed.  In such a case, I wouldn't say the tags are unrelated.

However, it is perfectly fine to delete tags (or branches or other refs) before running filter repo.  `git tag -d $tagname`.  (Or even `git update-ref -d refs/tags/$tagname`).  Or you can delete them after with the same commands.  Since commits don't belong to branches or tags, but rather branches and tags just point to some commit (and each commit points to its parent(s)), removing branches and tags is a lightweight change.
